Cpp_test/lib/ipc-queue-receive.cpp

```diff
@@ -0,0 +1,54 @@
+#include "ipc-queue-receive.h"
+#include <ctime>
+#include <iostream>
+
+ipcQueueReceive::~ipcQueueReceive()
+{
+	mq_close(this->mqd);
+	mq_unlink(this->info.method_name);
+}
+
+void ipcQueueReceive::init()
+{
+
+	std::cout << "Waiting for sender" << std::endl;
+	do
+	{
+		this->mqd = mq_open(this->info.method_name, O_RDONLY, 0660, &(this->attr));
+	} while (this->mqd == -1);
+
+	if (this->mqd == -1){
+		throw std::runtime_error("ERROR: Fail to open receive queue. ");
+	}
+	else{
+		std::cout << "/dev/mqueue" << this->info.method_name << " is opened." << std::endl;
+	}
+	this->file_handler.open_file();
+}
+
+void ipcQueueReceive::transfer()
+{
+	struct timespec	ts;
+	std::vector<char> buffer(this->attr.mq_msgsize);
+	long read_bytes{0};
+	std::cout << "Waiting for new file..." << std::endl;
+	errno = 0; // clear error number
+	
+	while (read_bytes != -1)
```

Comment:
Is there a reason for this instead of read_bytes>0 ?

Reply:
With read_bytes>0, program can not go into while loop

Reply:
Maybe you can initialize read_bytes with something else or you can do a `do while`

I asked this question because if there is some message with 0 data the while will continue.

Reply:
Ok then I will do with do while